packages/astro/src/core/config/settings.ts

```diff
@@ -134,3 +145,63 @@ export async function createSettings(config: AstroConfig, cwd?: string): Promise
 
 	return settings;
 }
+
+function validateTsconfig(settings: AstroSettings, logger: Logger, rawConfig: TSConfig) {
+	try {
+		if (!settings.config.experimental.tsconfig) {
+			return;
+		}
+
+		let newConfig = { ...rawConfig };
+
+		if (!rawConfig.extends) {
+			newConfig.extends = [GENERATED_TSCONFIG_PATH];
+			throw createTsconfigError(logger, rawConfig, newConfig);
+		} else if (
+			typeof rawConfig.extends === 'string' &&
+			rawConfig.extends !== GENERATED_TSCONFIG_PATH
+		) {
+			newConfig.extends = [rawConfig.extends, GENERATED_TSCONFIG_PATH];
+			throw createTsconfigError(logger, rawConfig, newConfig);
+		} else if (!rawConfig.extends.includes(GENERATED_TSCONFIG_PATH)) {
+			newConfig.extends = [...rawConfig.extends, GENERATED_TSCONFIG_PATH];
+			throw createTsconfigError(logger, rawConfig, newConfig);
+		}
+
+		if (rawConfig.include) {
+			logger.warn(
+				'types',
+				`Your root "tsconfig.json" has an "include" field. This will break types, please move it to your Astro config experimental.tsconfig.include option`,
```

Comment:
```suggestion
				`Your root "tsconfig.json" has an "include" field. Please move it to your Astro config experimental.tsconfig.include option to avoid breaking types.`,
```